File: AgqrPlayer4Tv/AgqrPlayer4Tv/Model/TimetableStoredDataset.cs
using System;
using System.Collections.Generic;

namespace AgqrPlayer4Tv.Model
{
    public class TimetableStoredDataset
    {
        public Dictionary<DayOfWeek, List<TimetableProgram>> Data { get; set; }
        public int Version { get; set; }
        public DateTime UpdatedAt { get; set; }

        public bool IsExpired
        {
            get { return (DateTime.Now - this.UpdatedAt).TotalHours > 24; }
        }
    }
}

File: AgqrPlayer4Tv/app/src/main/java/org/misuzilla/agqrplayer4tv/model/TimetableStoredDataset.kt
package org.misuzilla.agqrplayer4tv.model

import org.threeten.bp.DayOfWeek
import org.threeten.bp.Duration
import org.threeten.bp.LocalDateTime
import org.threeten.bp.ZoneOffset
import java.util.*

class TimetableStoredDataset(updatedAt: LocalDateTime) {
    val updatedAt = updatedAt.toEpochSecond(ZoneOffset.UTC)
    val version = 2

    val data = DayOfWeek.values().associate { it to mutableListOf<TimetableProgram>() }

    val isExpired: Boolean
        get() = LocalDateTime.ofEpochSecond(updatedAt + EXPIRE, 0, ZoneOffset.UTC) < LocalDateTime.now()

    companion object {
        val EXPIRE = Duration.ofHours(1).seconds

        val empty: TimetableStoredDataset by lazy {
            TimetableStoredDataset(LocalDateTime.MIN)
        }
    }
}